File: app/src/main/java/com/xephorium/crystalnote/data/model/Note.java
package com.xephorium.crystalnote.data.model;

import java.util.Date;

public class Note {

    private int color;
    private Date date;
    private String name;
    private String path;
    private String preview;

    public int getColor() {
        return color;
    }

    public void setColor(int color) {
        this.color = color;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getPreview() {
        return preview;
    }

    public void setPreview(String preview) {
        this.preview = preview;
    }
}

File: app/src/main/java/com/xephorium/crystalnote/data/model/Note.kt
package com.xephorium.crystalnote.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity(tableName = "notes")
data class Note(
    @PrimaryKey(autoGenerate = true) var id: Int = 0,
    var name: String,
    var contents: String,
    @ColumnInfo(defaultValue = "")
    var preview: String,
    var date: Date,
    var color: Int,
    var password: String = "",
    @ColumnInfo(defaultValue = "false")
    var archived: Boolean = false
) {
    companion object {
        const val NO_NOTE: Int = 0
    }
}
